ui/app/api/news/route.ts

```diff
@@ -0,0 +1,16 @@
+import { NextResponse } from "next/server";
+import fs from "node:fs/promises";
+import path from "node:path";
+
+export async function GET() {
+  try {
+    const dataDirectory = path.join(process.cwd(), "public", "data");
+    const filePath = path.join(dataDirectory, "index.json");
+    const fileContents = await fs.readFile(filePath, "utf8");
+    const data = JSON.parse(fileContents);
+    return NextResponse.json(data);
+  } catch (error) {
+    console.error("Error reading news data:", error);
```

Comment:
Better to add a single util to collect logs, such as define a `reporter` which contains the `log|info|warn|error` methods, so we can refactor it easier later or customize the behavior in different environment.